### PR TITLE
TEAMS-892 Fixed animation for mobile(added new ElementTransition util), fixed i…

### PR DESCRIPTION
**Pull request by IljaFatkulin**
…mage rendering on PDP, removed SharedTransition

**Related issue(s):**
* Fixes https://scandiflow.atlassian.net/browse/TEAMS-892

**Problem:**
* On mobile when using product cards to redirect on product page, image is transition with lags

**In this PR:**
* Added new ElementTransaction util where is fixed animation for mobile, deleted old - SharedTransition, changed image rendering on PDP

### CODE REVIEW

File: packages/scandipwa/src/component/ProductGallery/ProductGallery.style.scss [hunk 1]
```diff
@@ -35,8 +35,13 @@
     }
 
     @include mobile {
-        height: calc(100vh / 2);
-        margin-block-end: 16px;
+        // height: calc(100vh / 2);
```
Comment (Axector) [inline packages/scandipwa/src/component/ProductGallery/ProductGallery.style.scss:38]: Remove commented code.

File: packages/scandipwa/src/component/ProductGallery/ProductGallery.style.scss [hunk 2]
```diff
@@ -68,8 +73,9 @@
     }
 
     &-SliderWrapper {
-        flex: 1;
-        height: 100%;
+        // flex: 1;
+        // max-height: 100%;
```
Comment (Axector) [inline packages/scandipwa/src/component/ProductGallery/ProductGallery.style.scss:77]: Remove commented code.

File: packages/scandipwa/src/util/ElementTransition/ElementTransition.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    const translateY = tt - st;
+    const scale = tw / sw;
+    // const scaleY = th / sh;
+
+    // clonedElement.style.transform = `translate3d(${ translateX }px, ${ translateY }px, 0) scale(${ scaleX }, ${ scaleY })`;
+    clonedElement.style.transform = `translate3d(${ translateX }px, ${ translateY }px, 0) scale(${ scale })`;
+
+    const productGallery = document.querySelector('.ProductGallery');
+    const th = sh * scale;
+    productGallery.style.height = `${ th }px`;
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/util/ElementTransition/ElementTransition.js:87]: Preferable to remove this behavior

File: packages/scandipwa/src/util/ElementTransition/ElementTransition.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/scandipwa
+ * @link https://github.com/scandipwa/scandipwa
+ */
+
+export const transitionElementFromTo = async ({
```
Comment (Axector) [inline packages/scandipwa/src/util/ElementTransition/ElementTransition.js:13]: Move namespace closer to function.

File: packages/scandipwa/src/util/ElementTransition/ElementTransition.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    const {
+        top: tt, left: tl, width: tw,
+    } = imageTo.getBoundingClientRect();
+
+    const translateX = tl - sl;
+    const translateY = tt - st;
+    const scale = tw / sw;
+    // const scaleY = th / sh;
+
+    // clonedElement.style.transform = `translate3d(${ translateX }px, ${ translateY }px, 0) scale(${ scaleX }, ${ scaleY })`;
```
Comment (Axector) [inline packages/scandipwa/src/util/ElementTransition/ElementTransition.js:82]: Removed commented code.

**Review by Axector — CHANGES_REQUESTED** [review]
Check comments.

**Pull request merged**